src/types.rs

```diff
@@ -93,4 +93,8 @@ pub trait Interface: Send + Sync + InterfaceClone {
     fn get_call_stack(&self) -> Result<Vec<Address>> {
         bail!("unimplemented function get_call_stack in interface")
     }
+    fn generate_event(&self, event: String) -> Result<()> {
+        // TODO should be a SCEvent
+        bail!("unimplemented function generate_event in interface")
```

Comment:
Let's not bail, but succeed without doing anything. That way when we implement it there will be no breaking change.

Reply:
It also needs to be declared here I think: https://github.com/massalabs/sc-runtime/blob/feature/add_generate_event/src/execution_impl.rs#L29-L38

Reply:
and probably here as well: https://github.com/massalabs/sc-runtime/blob/feature/add_generate_event/src/abi_impl.rs

Reply:
From what I understand, we bail here only if the interface is not implemented for real somewhere else

Reply:
Ok for the bail :) you're right. The 2 other comments in this thread are important tho